src/runtime/cli/config/configuration-qemu.toml.in

```diff
@@ -274,7 +276,14 @@ valid_file_mem_backends = @DEFVALIDFILEMEMBACKENDS@
 # The value means the number of pcie_root_port
 # This value is valid when hotplug_vfio_on_root_bus is true and machine_type is "q35"
 # Default 0
-#pcie_root_port = 2
+pcie_root_port = 2
+
+# Delay in seconds to attach the large PCI Bar devices
+# Use this parameter when using some large PCI bar devices, and the guest kernel is 5.x
+# The value means the delay in seconds for attach a large PCI device
+# This value is valid when hotplug_vfio_on_root_bus is true, machine_type is "q35" and pcie_root_port is configured.
+# Default 0
+pcie_lazy_attach_delay = 3
 
```

Comment:
I think we should really mention here that this is a workaround for qemu/kernel behaviour, it's not something that really makes sense of itself.

Reply:
updated.

Reply:
Is it only me that thinks an arbitrary delay is risky? I understand it will take some time until the right solution is implemented and there is a need for a workaround, what about using the qemu-agent (does KATA use it?) to query the OS state or at least the port state?

Reply:
It's not just you.  I don't think Kata does use the qemu guest agent, but the more fundamental problem is that I'm not terribly clear on what we'd need to check for from guest userspace to be confident we're in the right state.

Reply:
We can query the PCIe Root Port(s) registers instead of "looking at" the Root Port blinking using a tool like setpci. Is far from perfect, but better than a "random" delay.

Reply:
I don't think that quite works.  The problem is that the state of the registers before and after the scan is the same "light is off".  From qemu, where we get the register updates directly, the latencies are low enough that we can probably get away with just not initiating hotplug when the light is blinking.

But once we introduce the latencies of sending messages to the agent and running tools like setpci, it's likely that we'll see the "light off" but it's started blinking by the time we actually start the hotplug.

Watching for it blinking then stopping also wouldn't be unreliable, because for the same reason we could just miss it and never think we're ready for the hotplug.

Reply:
@dgibson sorry for the misunderstanding. I meant this hack would work only as a workaround for the boot issue, not as a solution. For the boot sequence we can probe some registers of the pcie root port (including or not the blinking one) to understand if the port is "ready for the first time".

Reply:
@marcel-apf
> Is it only me that thinks an arbitrary delay is risky?
I limited the valid range of pcie_lazy_attach_delay, the valid value should be in [0,5]

> to query the OS state or at least the port state
It is better to hotplug the VFIO device after pciehp is ready.
I once considered getting the status of pciehp from the dmesg log.
Whatever, kata shim needs to communicate with the guest kernel through kata-agent.